source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -270,6 +275,43 @@ void AuthenticatorImpl::verifyKey() {
   handleGoodJwt(/*cache_hit=*/false);
 }
 
+void AuthenticatorImpl::addJWTClaimToHeader(const std::string& claim_name,
+                                            const std::string& header_name) {
+  StructUtils payload_getter(jwt_->payload_pb_);
+  const ::Envoy::ProtobufWkt::Value* claim_value;
+  const auto status = payload_getter.GetValue(claim_name, claim_value);
+  std::string str_claim_value;
+  if (status == StructUtils::OK) {
+    if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kStringValue) {
+      str_claim_value = claim_value->string_value();
+    } else if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kNumberValue) {
+      if (claim_value->number_value() < 0 ||
+          claim_value->number_value() >=
+              static_cast<double>(std::numeric_limits<uint64_t>::max())) {
+        ENVOY_LOG(debug, "--------claim : {} with value : {} not a valid integer -----------",
+                  claim_name, claim_value->number_value());
+      }
+      const uint64_t int_claim_value = static_cast<uint64_t>(claim_value->number_value());
+      str_claim_value = std::to_string(int_claim_value);
```

Comment:
We're doing a double -> uint64 -> string here? Seems potentially dangerous, i.e. the signed claim was on some double value, we are now forwarding it with some truncation of the floating portion. @qiwzhang WDYT?

Reply:
It is not dangerous, but we may have some precision loss from two fronts

1)  Using "double" in proto.Struct to represent JSON number,  but its [RFC](https://www.rfc-editor.org/rfc/rfc7159#section-6) say that it is ok.
2)  The code in this pr drop negative and fraction. 

For 2),  I will suggest to use std::to_string<double> directly.

Reply:
The only problem with this is we will never get integer output from jwt_verify library(https://github.com/google/jwt_verify_lib/blob/master/src/struct_utils.cc#L105). Because we never get int from google protobuf, it's of double type (https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.pb.h#L350). So when user has passed a integer (eg. 500), the `jwt-verify` returns something like (`500.0000`) which could be an issue. Generally claims are non negative integers, so it's better to stick to this. I can do one thing, if it is not an valid integer then just add it as a double value in the claim ? @qiwzhang @htuch wdyt ?

Reply:
added one solution, lmk what you guys think about it ?

Reply:
SG

Reply:
Can you folks explain for my benefit (and posterity), why it's safe to ignore the float portion? My naïve thinking here is that a JWT claim is about a specific value, e.g. "Joe Doe is allowed access to this document when the weather is warmer than 34.5 degrees celsius". If we ignore the float, Joe Doe gets access when the weather is 34 degrees celsius. Maybe this example is contrived, but unless there is some convention about discarding float portion or precision here, this seems a plausible attack vector when floating point claims are used.

Reply:
Sorry, we did not make it clear. We never ignore any fraction.   Here is what we do
```
   if   detect double is unsigned integer:  
          return std::to_string<uint64_t> ()
   else 
       return std::to_string<double>() 
```
If a number has fraction, we still print it as "double".   Just for nice print for integer, we do such optimization.
If we always use std::to_string<double>,  a "100" integer, will be "100.0000", this is not nice.